backend-implicaction/src/test/java/com/dynonuggets/refonteimplicaction/controller/JobsPostingControllerTest.java

```diff
@@ -126,4 +131,63 @@ void getJobByIdWithoutJwtShouldBeForbidden() throws Exception {
 
         verify(jobPostingService, times(0)).getJobById(anyLong());
     }
+
+    @Test
+    @WithMockUser
+    void archiveJobShouldChangeStatus() throws Exception {
+        JobPostingDto jobPostingDtoAvailable = JobPostingDto.builder()
+                .id(1L)
+                .createdAt(Instant.now())
+                .description("description")
+                .location("Paris")
+                .keywords("toto,yoyo")
+                .salary("1111$")
+                .status(statusAvailable)
+                .build();
+
+        JobPostingDto jobPostingDtoArchived = JobPostingDto.builder()
+                .id(1L)
+                .createdAt(Instant.now())
+                .description("description")
+                .location("Paris")
+                .keywords("toto,yoyo")
+                .salary("1111$")
+                .status(statusArchived)
+                .build();
+
+        given(jobPostingService.archiveOrUnarchiveJobPosting(jobPostingDtoAvailable.getId())).willReturn(jobPostingDtoArchived);
+
+        mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, jobPostingDtoAvailable.getId()).contentType(APPLICATION_JSON)).andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", is(Math.toIntExact(jobPostingDtoAvailable.getId()))))
```

Comment:
En fait j'me suis aperçu que tu pouvais simplifier comme ça : 
```suggestion
                .andExpect(jsonPath("$.id", is((jobPostingDtoAvailable.getId().intValue())))
```
https://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#intValue()